### PR TITLE
Unified layer dialog raster mods

### PR DESCRIPTION
**Pull request by elpaso**
## Description

This PR adds the two missing source select provider dialogs to OGR and GDAL and move some more functionality to the base abstract class.

**All providers now use the base abstract class and share the same loginc for buttons and signals/slot connections**

The OpenVectorFile dialog has now been moved from gui/ogr into the ogr provider directory where it belongs.

# New raster widget for the unified dialog

unified-dialog-raster

### CODE REVIEW

**Comment by nirvn** [comment]
This is great. You're doing an amazing job polishing up the unified layer dialog.

File: src/gui/qgsabstractdatasourcewidget.h [hunk 1]
```diff
@@ -86,6 +86,13 @@ class GUI_EXPORT QgsAbstractDataSourceWidget : public QDialog
     //! Emitted when a vector layer has been selected for addition
     void addVectorLayer( const QString &uri, const QString &layerName );
 
+    /** Emitted when one or more OGR supported layers are selected for addition
+     * \param layerQStringList list of layers protocol URIs
+     * \param enc encoding
+     * \param dataSourceType string (can be "file" or "database")
+     */
+    void addVectorLayers( const QStringList &layerQStringList, const QString &enc, const QString &dataSourceType );
```
Comment (nyalldawson) [inline src/gui/qgsabstractdatasourcewidget.h]: Can we change  `layerQStringList` to just `layerList` ? layerQStringList is a bit clumsy to me. Also enc should be encoding - see https://github.com/qgis/qgis3.0_api/issues/53

**Review by nyalldawson — APPROVED** [review]

File: src/providers/ogr/qgsogrsourceselect.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    void on_radioSrcFile_toggled( bool checked );
+    void on_radioSrcDirectory_toggled( bool checked );
+    void on_radioSrcDatabase_toggled( bool checked );
+    void on_radioSrcProtocol_toggled( bool checked );
+    void on_btnNew_clicked();
+    void on_btnEdit_clicked();
+    void on_btnDelete_clicked();
+    void on_cmbDatabaseTypes_currentIndexChanged( const QString &text );
+    void on_cmbConnections_currentIndexChanged( const QString &text );
+    void on_buttonBox_helpRequested() { QgsHelp::openHelp( QStringLiteral( "working_with_vector/supported_data.html#loading-a-layer-from-a-file" ) ); }
```
Comment (DelazJ) [inline src/providers/ogr/qgsogrsourceselect.h]: Could you replace this link by `managing_data_source/opening_data.html#loading-a-layer-from-a-file`

**Comment by DelazJ** [comment]
For consistency with other tabs, do we need the OK button?

**Comment by elpaso** [comment]
@DelazJ the ok/add buttons are now in all dialogs, the button bar is now configured in the base class. The advantage is that if we change our minds about the buttons we just need to change in a single place and that the look and feel of all the dialog is more consistent.

**Pull request merged**